clients/src/main/java/org/apache/kafka/common/serialization/Deserializer.java

```diff
@@ -60,6 +62,17 @@ default T deserialize(String topic, Headers headers, byte[] data) {
         return deserialize(topic, data);
     }
 
+    /**
+     * Deserialize a record value from a ByteBuffer into a value or object.
+     * @param topic topic associated with the data
+     * @param headers headers associated with the record; may be empty.
```

Comment:
sorry, not super familiar with the Headers class, assume empty is not null? It would be good to clarify here in case if user think empty = null.

Reply:
I copied this documentation from `deserialize(String topic, Headers headers, byte[] data)`.